cloud/parameterstore/fakeparameter/fake_parameter.go

```diff
@@ -42,3 +44,10 @@ func (p *FakeParameter) Insert(ctx context.Context) error {
 	_, err := evergreen.GetEnvironment().DB().Collection(Collection).InsertOne(ctx, p)
 	return err
 }
+
+// Upsert inserts a single parameter into the fake parameter store or replaces
+// an one if one with the same ID already exists.
+func (p *FakeParameter) Upsert(ctx context.Context) error {
+	_, err := evergreen.GetEnvironment().DB().Collection(Collection).ReplaceOne(ctx, bson.M{IDKey: p.ID}, p, options.Replace().SetUpsert(true))
```

Comment:
nit: can we use `db.Upsert` here instead?

Reply:
I wouldn't say the db package is officially deprecated, although if we do want to pursue [DEVPROD-53](https://jira.mongodb.org/browse/DEVPROD-53) I agree we shouldn't be using it in more places.

Reply:
If we're interested in keeping the intermediate helper functions, then yeah, we could keep the db package. But for now, the functions aren't migrated to use the context and we have a number of places that directly use the mongo driver as the workaround, so I feel like this discussion should be deferred to future work when the db helper package is updated.